### PR TITLE
std.ArrayList.unusedCapacitySlice: Return unaligned slice

### LINKED ISSUE #19724: Compile error using `unusedCapacitySlice` with `ArrayListAligned` if alignment > 1
### Zig Version

0.13.0-dev.4+c7ffdbcd4

### Steps to Reproduce and Observed Behavior

```zig
const std = @import("std");
test {
    var list = std.ArrayListAligned(u8, 4).init(std.testing.allocator);
    defer list.deinit();
    try list.appendNTimes(1, 4);
    _ = list.unusedCapacitySlice();
}
```

```
local/zig-macos-x86_64-0.13.0-dev.4+c7ffdbcd4/lib/std/array_list.zig:570:41: error: expected type '[]align(4) u8', found '[]u8'
            return self.allocatedSlice()[self.items.len..];
                   ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
local/zig-macos-x86_64-0.13.0-dev.4+c7ffdbcd4/lib/std/array_list.zig:570:41: note: pointer alignment '1' cannot cast into pointer alignment '4'
local/zig-macos-x86_64-0.13.0-dev.4+c7ffdbcd4/lib/std/array_list.zig:569:48: note: function return type declared here
        pub fn unusedCapacitySlice(self: Self) Slice {
                                               ^~~~~
referenced by:
    test_0: test.zig:6:13
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
```

### Expected Behavior

Function should return either return an unaligned slice, or assert correct alignment and return a slice with the same alignment as the array list.

### PR DESCRIPTION
**Pull request by gooncreeper**
Fixes #19724

### CODE REVIEW

**Comment by nektro** [comment]
`[]T` means natural alignment, not no alignment. potential patch should include the test from the issue added to the file

**Comment by gooncreeper** [comment]
> `[]T` means natural alignment, not no alignment. potential patch should include the test from the issue added to the file

Do you think a test is really necessary for a semantic error? If so I suppose I could add one.
Also unaligned in the sense of ArrayListAligned

**Comment by nektro** [comment]
> Do you think a test is really necessary for a semantic error? If so I suppose I could add one.

these are the most important kind. we want to make sure that if the code changes in the future this behavior does not regress.

> Also unaligned in the sense of ArrayListAligned

gotcha, yeah makes sense reading the issue again now

File: lib/std/array_list.zig [hunk 1]
```diff
@@ -2242,3 +2242,10 @@ test "return OutOfMemory when capacity would exceed maximum usize integer value"
         try testing.expectError(error.OutOfMemory, list.ensureUnusedCapacity(2));
     }
 }
+
+test "ArrayListAligned with non-native alignment compiles unusedCapabitySlice" {
+    var list = ArrayListAligned(u8, 4).init(testing.allocator);
+    defer list.deinit();
+    try list.appendNTimes(1, 4);
+    _ = list.unusedCapacitySlice();
```
Comment (andrewrk) [inline lib/std/array_list.zig:2250]: for future reference, tests should use the API, not only make sure it compiles.

**Pull request merged**